torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -294,100 +234,194 @@ def _wrap_fx_args_as_onnxscript_args(
     return onnxscript_args, onnxscript_kwargs
 
 
-@_beartype.beartype
-@diagnostics.diagnose_call(
-    diagnostics.rules.fx_node_to_onnx,
-    diagnostic_message_formatter=_fx_node_to_onnx_message_formatter,
-)
-def _export_fx_node_to_onnxscript(
-    diagnostic_context: diagnostics.DiagnosticContext,
-    node: torch.fx.Node,
-    onnxscript_graph: graph_building.TorchScriptGraph,
-    fx_name_to_onnxscript_value: Dict[
-        str,
-        Union[
-            graph_building.TorchScriptTensor,
-            Tuple[graph_building.TorchScriptTensor, ...],
-        ],
-    ],
-    tracer: graph_building.TorchScriptTracingEvaluator,
-    fx_module_with_metadata: torch.fx.GraphModule,
-    options: ResolvedExportOptions,
-):
-    # Record stack trace of node in diagnostic.
-    node_stack_trace = node.stack_trace
-    if node_stack_trace:
-        diagnostic = diagnostic_context.inflight_diagnostic(
-            rule=diagnostics.rules.fx_node_to_onnx
-        )
-        diagnostic.with_additional_message(
-            f"### PyTorch source information\n```\n{node_stack_trace}\n```"
+class FxOnnxInterpreter:
```

Comment:
This class is stateful. Let's define all methods as `staticmethod` so that we can maintain as much stateless behavior as possible.

Reply:
> I prefer stateless code as well, but the thing is that the current code is stateful. What changed was that before we had a stateful helper, now we have a stateful class.
> 

Sorry but I disagree. They are not stateful as long as they don't interact with `self`.

> I'd prefer having a statefulness refactoring on a separate PR to not add unrelated complexity to an already big PR

Well, the major concern I have is the use of `self`. The current codebase carefully avoids it so introducing `self` is a huge architectural change and should be discussed before making a PR. To move forward, instead of implementing a stateful class, you can move interpreter methods into this class as `staticmethod`s.

Reply:
will do